src/State/Hub.php

```diff
@@ -169,6 +172,31 @@ public function captureLastError(?EventHint $hint = null): ?EventId
         return null;
     }
 
+    public function captureCheckIn(string $slug, MonitorConfig $upsertMonitorConfig, CheckInStatus $status, ?CheckIn $previous = null): ?CheckIn
+    {
+        $client = $this->getClient();
+
+        if (!$client instanceof ClientInterface) {
```

Comment:
I would find more appropriate to check that the client is not `null` because the the real meaning of this line is to ensure that that instance is not `null` and not that it implements the interface. It is a small change indeed, but imo it conveyes to the reader the expectation in a clearer way.